x-pack/plugins/security/server/routes/api_keys/create.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+import { ApiKey } from '../../../common/model';
+import { wrapError, wrapIntoCustomErrorResponse } from '../../errors';
+import { RouteDefinitionParams } from '..';
+
+interface ResponseType {
+  itemsInvalidated: Array<Pick<ApiKey, 'id' | 'name'>>;
+  errors: Array<Pick<ApiKey, 'id' | 'name'> & { error: Error }>;
+}
+
+export function defineCreateApiKeyRoutes({ router, clusterClient }: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/api_key',
+      validate: {
+        body: schema.object({
+          name: schema.string(),
+          expiration: schema.maybe(schema.string()),
```

Comment:
Did you intentionally leave out `role_descriptors` here?

Reply:
Good catch :)